src/ndarray/ndarray.cc

```diff
@@ -1214,6 +1214,31 @@ void NDArray::SyncCopyToCPU(void *data, size_t size) const {
   }
 }
 
+void NDArray::CheckFormat(const bool full_check) const {
+  NDArray cpu_ret = NDArray(mshadow::Shape1(1), Context::CPU());
+  auto err = cpu_ret.data().dptr<mshadow::default_real_t>();
+  *err = 0;
```

Comment:
Is `err` already set in `CheckFormatImpl `? Does `CheckFormatImpl` rely on input value being initialized to a default value?

Reply:
No, this line should be removed.